File: Problemset/longest-common-prefix/longest-common-prefix.go

// @Title: 最长公共前缀 (Longest Common Prefix)
// @Author: 846188037@qq.com
// @Date: 2021-03-18 09:59:58
// @Runtime: 0 ms
// @Memory: 2.3 MB

func longestCommonPrefix(strs []string) string {
    if len(strs)==0{
        return ""
    }
    prefix:=strs[0]
    for _,v := range strs{
        for i:=0;i<len(prefix);i++{
            if i==len(v){
                prefix=v
                break
            }
            if prefix[i]!=v[i]{
                prefix=prefix[:i]
                break
            }
        }
    }
    return prefix

}

File: Problemset/longest-common-prefix/longest-common-prefix.js

// @Title: 最长公共前缀 (Longest Common Prefix)
// @Author: 846188037@qq.com
// @Date: 2020-06-20 09:04:58
// @Runtime: 116 ms
// @Memory: 35.6 MB

/**
 * @param {string[]} strs
 * @return {string}
 */
var longestCommonPrefix = function(strs) {
    var a=""
    if(strs.length==0 || strs[0].length==0)
    return a;
    for(var k=0;k<strs[0].length;k++)
    {
        for(var i=0;i<strs.length;i++)
        {
        if (strs[i].charAt(k)!=strs[0].charAt(k))
        return a;
        }
       
        a=a+strs[0].charAt(k);
        

    }
    return a;
    

};
